client/src/templates/Challenges/projects/frontend/show.tsx

```diff
@@ -67,165 +67,120 @@ interface ProjectProps {
   updateSolutionFormValues: () => void;
 }
 
-// Component
-class Project extends Component<ProjectProps> {
-  static displayName: string;
-  private container: React.RefObject<HTMLElement> = React.createRef();
+const ShowFrontEndProject = (props: ProjectProps) => {
+  const handleSubmit = ({
+    showCompletionModal
+  }: {
+    showCompletionModal: boolean;
+  }): void => {
+    if (showCompletionModal) {
+      props.openCompletionModal();
+    }
+  };
 
-  constructor(props: ProjectProps) {
-    super(props);
-    this.handleSubmit = this.handleSubmit.bind(this);
-  }
-  componentDidMount() {
+  const container = useRef<HTMLElement>(null);
+
+  useEffect(() => {
     const {
       challengeMounted,
       data: {
         challengeNode: {
-          challenge: {
-            fields: { tests },
-            title,
-            challengeType,
-            helpCategory
-          }
+          challenge: { fields, title, challengeType, helpCategory }
         }
       },
       pageContext: { challengeMeta },
       initTests,
       updateChallengeMeta
-    } = this.props;
-    initTests(tests);
+    } = props;
+    initTests(fields.tests);
     updateChallengeMeta({
       ...challengeMeta,
       title,
       challengeType,
       helpCategory
     });
     challengeMounted(challengeMeta.id);
-    this.container.current?.focus();
-  }
+    container.current?.focus();
+  }, [props]);
```

Comment:
```suggestion
    // This effect should be run once on mount
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
```
It should never happen that the props change, so we don't need to include them as deps. Also, if they _did_ change, that would cause the hotkeys to receive focus (which we don't want).